### PR TITLE
feat: add unit test for types.ts

### LINKED ISSUE #371: [TEST] Create unit tests `for utils/scaffold-stark/types.ts` file
## Issue Overview

Unit tests required for the indicated file to ease future refactoring and feature validations.

## Proposed Solutions or Ideas (more like tips and reminders)

- We use vitest to run the unit tests, can refer to their documentation for the syntax: https://vitest.dev/, Jest developers shouldn't have too much of a hard time getting used to the syntax.
- Note that hooks testing tools are also now imported via `@testing-library/react`,
- Please refer to `README.md` on how to execute the tests
- Use `it.only()` to isolate tests in case you need to, but do not forget to remove it once you are submitting your PR.
- Mock when necessary. please make sure that the tests *NOT ONLY* works on your device
- If you're not sure of the features of the code unit to test, please refer to the [documentation](https://docs.scaffoldstark.com/hooks/), or free to ask on this issue page or on the Telegram Group

## Code Coverage
*IMPORTANT* - When submitting a PR, please provide a screenshot of the coverage for the specific units you are working on. We would like to achieve >= 70 % code coverage. 

Happy Hacking!

Example: 
image

### PR DESCRIPTION
**Pull request by coxmars**
# Unit test for types.ts

- Fixes #371

## Types of change

- [ ] Feature
- [ ] Bug
- [X] Enhancement

## Comments 

Added comprehensive unit tests for utils/scaffold-stark/types.ts file with:

### Unit tests for all type checking functions

- [x] Cairo integer types (u8, u16, u32, etc.)
- [X] Cairo big integer types (u64, u128)
- [X] Special types (U256, ContractAddress, EthAddress, etc.)
- [X] Tuple types
- [X] Struct and enum type checks
- [X] Collection types (Array, Option, Result)
- [X] Generic type parsing

### Test coverage includes

- [X] Positive cases (valid type checks)
- [X] Negative cases (invalid types)
- [X] Edge cases and type variants

### Test execution and coverage

Screenshot 2024-12-17 at 8 49 16 AM

Screenshot 2024-12-17 at 8 49 58 AM

### CODE REVIEW

File: packages/nextjs/utils/scaffold-stark/__test__/types.test.ts [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+  });
+
+  describe("isStructOrEnum", () => {
+    it("should return true for struct types", () => {
+      expect(isStructOrEnum({ type: "struct" })).toBe(true);
+    });
+
+    it("should return true for enum types", () => {
+      expect(isStructOrEnum({ type: "enum" })).toBe(true);
+    });
```
Comment (metalboyrick) [inline packages/nextjs/utils/scaffold-stark/__test__/types.test.ts:118]: can we have a real looking example, would be better IMO

image

**Review by metalboyrick — CHANGES_REQUESTED** [review]
nice work ser, thank you so much! minor comments

**Review by metalboyrick — APPROVED** [review]
Looks good to me!
